test/release/examples/primers/arrays.chpl

```diff
@@ -1,45 +1,56 @@
 // Arrays
 
 //
-// This primer is a tutorial on Chapel rectangular arrays and domains.
+// This primer is a tutorial on Chapel's rectangular arrays and
+// domains.  Other primers cover Chapel's :ref:`associative
+// <primers-associative>` and :ref:`sparse <primers-sparse>` arrays,
+// building on concepts introduced here.
 //
 
 //
 // Arrays in Chapel are specified using a square-bracketed expression
-// that specifies the array's index set, followed by the array's
-// element type.  Rectangular arrays are those whose indices are
-// integers or tuples of integers, supporting standard
-// multidimensional, rectilinear index sets.
+// that specifies the array's index set or *domain*, followed by the
+// array's element type.  Rectangular arrays are those whose indices
+// are integers or tuples of integers, bounded by a range in each
+// dimension, supporting multidimensional, rectilinear index sets.
 //
 
+// Declaring Arrays
+// ----------------
+
+// Let's start by declaring an ``n``-element array of 64-bit ``real``
+// values (where ``n`` defaults to 5):
+
 config const n = 5;
 
-//
-// Declare an array of five 64-bit real values:
-//
 var A: [1..n] real;
 
 //
 // Like other variable types in Chapel, arrays are initialized so that
-// each element stores its default value.  So our array of real values
-// above will default to an array of the value 0.0.
+// each element stores its default value.  So our array of ``real`` values
+// above will default to an array whose elements each store the value
+// 0.0.
 //
 
-writeln("Initially, A is: ", A);
+writeln("Initially, A is: ", A);  // prints 0.0 for each array element
 
 //
-// Arrays can also be declared using the array literal syntax.
-// Array literals are specified by enclosing a comma separated list of
-// expressions in square brackets.  The domain of the array will be
-// 0-based, and the type of the array's elements will be that of the
-// expressions if they all have the same type.  If they do not, the
-// array's element type will be a type that can hold all of the values
-// if the compiler can determine that one exists.
+// Arrays can also be declared using array literals.  These are
+// specified by enclosing a comma-separated list of expressions in
+// square brackets.  The domain of the array will be 0-based, and the
```

Comment:
Maybe: "Unless otherwise specified, the domain of the array will be 0-based"?

Reply:
Hmm, good point.  In writing this, I was thinking that I was describing the domain of the array literal itself, but I see that it's ambiguous, and probably leaning more toward the variable if anything.  I've tried to clarify and make it more precise by changing to:

> // ... Unless otherwise specified, the domain of the new          
> // array variable will be 0-based, and the type of its elements will            
> // be that of the expressions if they all have the same type.